src/pages/reviews/all.tsx

```diff
@@ -0,0 +1,226 @@
+import {
+  Button,
+  Flex,
+  Heading,
+  Input,
+  Menu,
+  MenuButton,
+  MenuItem,
+  MenuList,
+  Text,
+} from "@chakra-ui/react";
+import { useSession } from "next-auth/react";
+import { BiChevronDown } from "react-icons/bi";
+import { IoIosFunnel } from "react-icons/io";
+import AuthStatus from "@/components/transcript/AuthStatus";
+import { useRouter } from "next/router";
+import { FilterQueryNames, ReviewStatus } from "@/config/default";
+import Pagination from "@/components/tables/Pagination";
+import { useDebouncedCallback } from "use-debounce";
+import { UI_CONFIG } from "@/config/ui-config";
+import AdminReviewsTable from "@/components/tables/AdminReviewsTable";
+import { useGetAllReviews } from "@/services/api/admin/useReviews";
+import { RefetchButton } from "@/components/tables/TableItems";
+
+// eslint-disable-next-line no-unused-vars
+type OnSelect<T> = (name: string, item: T) => void;
+
+const FilterItem = ({ text }: { text: string }) => (
+  <Flex userSelect={"none"} gap={1} alignItems={"center"}>
+    <Text textTransform="capitalize" fontWeight="bold" fontSize="14px">
+      {text}
+    </Text>
+    <BiChevronDown />
+  </Flex>
+);
+
+const SelectFilter = <T extends string>({
+  hasValue,
+  onSelect,
+  options,
+  name,
+}: {
+  hasValue: boolean;
+  onSelect: OnSelect<T>;
+  options: T[];
+  name: string;
+}) => {
+  return (
+    <Menu>
+      <MenuButton color={hasValue ? "orange.500" : undefined}>
+        <FilterItem text={name} />
+      </MenuButton>
+      <MenuList>
+        {options.map((item) => (
+          <MenuItem
+            textTransform={"capitalize"}
+            fontSize="12px"
+            fontWeight={"medium"}
+            onClick={(e) => {
+              e.stopPropagation();
+              onSelect(name, item);
+            }}
+            key={item}
+          >
+            {item}
+          </MenuItem>
+        ))}
+      </MenuList>
+    </Menu>
+  );
+};
+
+const Transactions = () => {
```

Comment:
if this component is the same as the `Transactions` component, we should make it dynamic to avoid duplicate code